src/renderer/graph/FrequencyResponseChart.tsx

```diff
@@ -16,45 +25,102 @@ const isFilterEqual = (f1: IFilter, f2: IFilter) => {
   );
 };
 
+interface IGraphData {
+  chartData: IChartCurveData[];
+  autoPreAmpValue: number;
+}
+
 const FrequencyResponseChart = () => {
   const { filters, preAmp, isAutoPreAmpOn, setPreAmp, isGraphViewOn } =
     useAquaContext();
   const prevFilters = useRef<IFilter[]>([]);
-  const prevFilterLines = useRef<ChartDataPoint[][]>([]);
+  const prevFilterLines = useRef<IChartLineDataPointsById>({});
+
+  const { chartData, autoPreAmpValue }: IGraphData = useMemo(() => {
+    const pointById: { [id: string]: IChartPointData } = {};
+
+    // Identify the index of each filter by id from previous render
+    const prevIndices: { [id: string]: number } = {};
+    prevFilters.current.forEach((f, i) => {
+      prevIndices[f.id] = i;
+    });
+
+    const updatedFilterLines: IChartLineDataPointsById = {};
 
-  const {
-    chartData,
-    autoPreAmpValue,
-  }: { chartData: ChartData[]; autoPreAmpValue: number } = useMemo(() => {
     // Update filter lines that have changed
-    const updatedFilterLines = filters.map((f, index) =>
-      index < prevFilters.current.length &&
-      isFilterEqual(f, prevFilters.current[index])
-        ? prevFilterLines.current[index]
-        : getFilterPoints(f)
-    );
+    filters.forEach((filter) => {
+      // Store control point info
+      pointById[filter.id] = { x: filter.frequency, y: filter.gain };
+
+      // New filters have no previous data
+      const prevIndex = prevIndices[filter.id];
```

Comment:
Ideally, filters should be of type `{ [id: string]: IFilter}` instead of `IFilter[]`. Then, we can avoid creating this prevIndices array. Refactoring all instances of IFilter[] is going to take some work though so I think this is fine for now. Once we get jat-30, -90 -83, in ill start the refactor work for IFilter.

Reply:
Something related to this that we'll need to keep in mind is that we're currently assigning colors to filter curves based on the index in the sorted array. By coming up with a way to assign colors based on id, this would actually be even better cuz that would fix the issue of a filter's curve color changing when its position in the array changes.